plat/common/include/uk/plat/common/common.lds.h

```diff
@@ -136,6 +136,14 @@
 		KEEP(*(SORT_BY_NAME(.uk_event_*)))			\
 	}
 
+#define LIBPARAM_SECTION						\
+	uk_libparam_params_start = .;					\
```

Comment:
Following some weird paranormal events with `ld` misplacing some of these ld script defined syms in the case of empty sections, I think we should start keeping these defined within the scope of the section they are meant for:
```C
#define LIBPARAM_SECTION						\
	.uk_libparam_params :						\
	{								\
		uk_libparam_params_start = .;				\
		KEEP(*(SORT_BY_NAME(.uk_libparam_param*)))		\
		uk_libparam_params_end = .;				\
	}
```

Reply:
Hmmm, I'm not much of a believer of paranormal phenomena. Can you elaborate on the problem? Either way, if we make such a change, we should probably make it to all sections in that file on a follow-up PR.

Reply:
Update: I moved the section into a library-defined linker script, and moved the variables into the section. This also fixes a bug where the libukparam section would be mapped as RO. My proposal for a global update on `common.lds.h` holds.

Reply:
Update2: See discussion in https://github.com/unikraft/unikraft/pull/1471#discussion_r1694429633.